### PR TITLE
only show "Add to your Nextcloud" on share link if federation is acti…

### PR DESCRIPTION
**Pull request by violoncelloCH**
…vated

This only hides "Add to your Nextcloud" if the checkmark for federated sharing in settings is unchecked. Deactivating the federation app does not hide these settings and does not hide the "Add to your Nextcloud"-option as well.

fixes #12395

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareController.php [hunk 1]
```diff
@@ -455,14 +455,26 @@ public function showShare($path = ''): TemplateResponse {
...
-			if ($shareIsFolder) {
+			if ($shareIsFolder && $this->federatedShareProvider->isOutgoingServer2serverShareEnabled()) {
 				$response->setHeaderActions([
 					new SimpleMenuAction('download', $this->l10n->t('Download all files'), 'icon-download-white', $shareTmpl['downloadURL'], 0),
 					new SimpleMenuAction('download', $this->l10n->t('Download all files'), 'icon-download', $shareTmpl['downloadURL'], 10, $shareTmpl['fileSize']),
 					new LinkMenuAction($this->l10n->t('Direct link'), 'icon-public', $shareTmpl['previewURL']),
 					new ExternalShareMenuAction($this->l10n->t('Add to your Nextcloud'), 'icon-external', $shareTmpl['owner'], $shareTmpl['displayName'], $shareTmpl['filename']),
 				]);
-			} else {
+			} elseif ($shareIsFolder && !$this->federatedShareProvider->isOutgoingServer2serverShareEnabled()) {
```
Comment (juliushaertl) [inline apps/files_sharing/lib/Controller/ShareController.php]: Let's not duplicate all those header actions. Istead please build an array with those and only add the ExternalShareMenuAction if isOutgoingServer2serverShareEnabled() is true.

**Review by juliusknorr — DISMISSED** [review]

**Comment by violoncelloCH** [comment]
@juliushaertl like this? ^

**Comment by violoncelloCH** [comment]
@juliushaertl @ChristophWurst @skjnldsv can we have another review here?

**Review by skjnldsv — APPROVED** [review]
Much cleaner :)

**Comment by violoncelloCH** [comment]
may I add myself as author to the file header now? or what do you consider a "substantial change"?

**Comment by skjnldsv** [comment]
> may I add myself as author to the file header now? or what do you consider a "substantial change"?

You can always add yourself to the header. I'd even say you always should. But most of the time people forget :/

**Comment by violoncelloCH** [comment]
> You can always add yourself to the header. I'd even say you always should. But most of the time people forget :/

good to know! I've done it now...

**Comment by danielkesselberg** [comment]
```
$responseComposer = [];

if ($shareIsFolder) {
	$responseComposer[] = new SimpleMenuAction('download', $this->l10n->t('Download all files'), 'icon-download-white', $shareTmpl['downloadURL'], 0);
	$responseComposer[] = new SimpleMenuAction('download', $this->l10n->t('Download all files'), 'icon-download', $shareTmpl['downloadURL'], 10, $shareTmpl['fileSize']);
} else {
	$responseComposer[] = new SimpleMenuAction('download', $this->l10n->t('Download'), 'icon-download-white', $shareTmpl['downloadURL'], 0);
	$responseComposer[] = new SimpleMenuAction('download', $this->l10n->t('Download'), 'icon-download', $shareTmpl['downloadURL'], 10, $shareTmpl['fileSize']);
}

$responseComposer[] = new LinkMenuAction($this->l10n->t('Direct link'), 'icon-public', $shareTmpl['previewURL']);
if ($this->federatedShareProvider->isOutgoingServer2serverShareEnabled()) {
	$responseComposer[] = new ExternalShareMenuAction($this->l10n->t('Add to your Nextcloud'), 'icon-external', $shareTmpl['owner'], $shareTmpl['displayName'], $shareTmpl['filename']);
}
```

I think it looks cleaner if you do it like above. Yes the download buttons looks similar but they are different. If you dont agree just ignore :rofl:

**Review by kesselb — APPROVED** [review]
Well done :+1:

**Comment by violoncelloCH** [comment]
@danielkesselberg thank you! I get your point; the variables are only assigned and use once, so not really necessary -- however I think it's easier to read the code with the variables... But if you want the code to be more compact I can change it :)

**Comment by danielkesselberg** [comment]
> @danielkesselberg thank you! I get your point; the variables are only assigned and use once, so not really necessary -- however I think it's easier to read the code with the variables... But if you want the code to be more compact I can change it :)

Up to you ;-) If someone approves a pull request and suggests changes they are not mandatory (i think they are not mandatory but could be wrong :rofl:)

**Comment by violoncelloCH** [comment]
what's wrong with drone here?

File: apps/files_sharing/lib/Controller/ShareController.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
  * @author Björn Schießle <bjoern@schiessle.org>
  * @author Georg Ehrke <oc.list@georgehrke.com>
  * @author Joas Schilling <coding@schilljs.com>
+ * @author Jonas Sulzer <jonas@violoncello.ch>
```
Comment (skjnldsv) [inline apps/files_sharing/lib/Controller/ShareController.php]: :tada:

**Comment by skjnldsv** [comment]
@violoncelloCH can you rebase? Drone will recreate itself :)

**Comment by violoncelloCH** [comment]
rebase done...

**Comment by violoncelloCH** [comment]
looks like there is always one randomly failing drone test... :/

**Comment by MorrisJobke** [comment]
They are unrelated - it was a codecov failure.

**Pull request merged**